compiler/lexer/lexinput.c

```diff
@@ -0,0 +1,143 @@
+#include "lexinput.h"
+
+struct LEXINPUT {
```

Comment:
There's no reason for both `FILE *` and `ByteBuffer *bb; int pos; int iseof;` since we'll only ever use 1 at a time, so these two can go in a union.